dev/run-tests.py

```
@@ -0,0 +1,417 @@
+        # remove any empty strings
+        changed_files = [f for f in raw_output.split('\n') if f]
+
+        # find any sql files
+        sql_files = [f for f in changed_files
+                     if multi_starts_with(f, 
+                                          "sql/", 
+                                          "bin/spark-sql", 
+                                          "sbin/start-thriftserver.sh")]
+
+        non_sql_files = set(changed_files).difference(set(sql_files))
+
+        if non_sql_files:
+            test_suite.append("CORE")
+        if sql_files:
+            print "[info] Detected changes in SQL. Will run Hive test suite."
+            test_suite.append("SQL")
+            if not non_sql_files:
+                print "[info] Detected no changes except in SQL. Will only run SQL tests."
+        return set(test_suite)
+    else:
+        # we aren't in the Amplab environment so merely run all tests
+        test_suite.append("CORE")
+        test_suite.append("SQL")
+        return set(test_suite)
+
+
+def set_title_and_block(title, err_block):
+    os.environ["CURRENT_BLOCK"] = error_codes[err_block]
+    line_str = "".join(['='] * 72)
+
+    print
+    print line_str
+    print title
+    print line_str
+
+
+def run_apache_rat_checks():
+    set_title_and_block("Running Apache RAT checks", "BLOCK_RAT")
+    run_cmd(["./dev/check-license"])
+
+
+def run_scala_style_checks():
+    set_title_and_block("Running Scala style checks", "BLOCK_SCALA_STYLE")
+    run_cmd(["./dev/lint-scala"])
+
+
+def run_python_style_checks():
+    set_title_and_block("Running Python style checks", "BLOCK_PYTHON_STYLE")
+    run_cmd(["./dev/lint-python"])
+
+
+def exec_maven(mvn_args=[]):
+    """Will call Maven in the current directory with the list of mvn_args passed
+    in and returns the subprocess for any further processing"""
+
+    return subprocess.Popen(["./build/mvn"] + mvn_args)
```

Is there any reason to not just `wait()` here? It seems like that's what we're always doing with the return value.

Also, if we want to raise an exception when this fails (which I think we do), we should change `Popen()` to `check_call()`, in which case no need for the `wait()` either.